scripts/performance/Perf_runner.py

```diff
@@ -1,118 +1,374 @@
+        print("\n Lowest transaction (individual thread): {} second(s)".
+              format(str(self.lowest)), file=result_file)
+        print("\n Transaction per client: " + str(int(txns_per_client)),
+              file=result_file)
+        print("\n Total requested transactions: " + str(int(ttl_txns)),
+              file=result_file)
+        print("\n Total passed transactions: " + str(self.passed_req),
+              file=result_file)
+        print("\n Total failed transactions: " + str(self.failed_req),
+              file=result_file)
+        print("\n Average time of a transaction "
+              "(multiple threads): {} second(s)".
+              format(str((self.finish_time - self.start_time) / ttl_txns)),
+              file=result_file)
+        print("\n Estimated transactions per second: " + str(txns_per_second),
+              file=result_file)
+
+    def find_lowest_and_fastest_transaction(self):
+        """
+        Find lowest and fastest transactions.
+        """
+        self.lowest = self.list_tester[0].lowest_txn
+        self.fastest = self.list_tester[0].fastest_txn
+        for tester in self.list_tester:
+            temp_lowest = tester.lowest_txn
+            temp_fastest = tester.fastest_txn
+            if self.lowest < temp_lowest:
+                self.lowest = temp_lowest
+            if self.fastest > temp_fastest:
+                self.fastest = temp_fastest
+
+    def find_start_and_finish_time(self):
+        """
+        Find the earliest time that a client is started and latest time that a
+        client is finished.
+        """
+        self.start_time = self.list_tester[0].start_time
+        self.finish_time = self.list_tester[0].finish_time
+
+        for tester in self.list_tester:
+            if self.start_time > tester.start_time:
+                self.start_time = tester.start_time
+
+            if self.finish_time < tester.finish_time:
+                self.finish_time = tester.finish_time
+
+    def start_tester_in_thread(self):
+        """
+        Create thread and start all the tester in list.
+        """
+        threads = list()
+        for _ in range(self.options.clients):
+            tester = self.create_tester()
+            self.list_tester.append(tester)
+            thread = threading.Thread(target=self.run_tester_in_thread,
+                                      kwargs={'tester': tester})
+            thread.daemon = True
+            thread.start()
+            threads.append(thread)
+
+        for thread in threads:
+            thread.join()
+
+    @staticmethod
+    def run_tester_in_thread(tester):
+        """
+        Execute testing function of tester.
+        """
+        loop = asyncio.new_event_loop()
+        utils.run_async_method(loop, tester.test)
+        loop.close()
+
+    def create_tester(self):
+        """
+        Create tester base mode "-a", "-t", "-g", "-l".
+
+        :return: tester
+        """
+        if self.options.adding:
```

Comment:
Why do we need multiple scripts that all do almost the same? What is the difference between PerformanceTesterForAddingRequest, TesterSimulateLoad, TesterSimulateTraffic? Why can't we have just one script with parameters specifying type of requests, delay between requests, flag whether to wait for reply before sending the next request, etc.